src/main/kotlin/frc/robot/subsystems/ExampleSubsystem.kt

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+package frc.robot.subsystems
+
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+
+/** Creates a new ExampleSubsystem.  */
+class ExampleSubsystem(private val exampleProperty: Double): SubsystemBase() {
```

Comment:
Make this an `object` to show the singleton pattern

Reply:
What do you think about adding another example subsystem that's a singleton instead to show that u can do both ways

Reply:
Subsystems should **not** be singletons imo.

It encourages messy resource management and bandaid fixes instead of a deliberate flow of data/actions.

It's ok if you want to do this in your own code but using in an example seems foolish.

Even falon voiced issues they had this year with subsystems as singletons. Yes you only want 1 of each subsystem but that does not make them a "singleton" in the context of this scenario.

Reply:
> what is the loading/initialization semantics of Kotlin singletons?

iirc they load once you reference them.


IMO its fine to make some subsystems singletons and its a pattern that should be shown to allow people to make informed decisions for themselves